modules/Behaviour/behaviour_manage_add.php

```diff
@@ -203,20 +204,15 @@
 
                     //Lessons
                     $lessons = array();
-                    $minDate = date('Y-m-d', (time() - (24 * 60 * 60 * 30)));
 
-                        $dataSelect = array('date1' => date('Y-m-d', time()), 'date2' => $minDate, 'gibbonSchoolYearID' => $session->get('gibbonSchoolYearID'), 'gibbonPersonID' => $values['gibbonPersonID']);
-                        $sqlSelect = "SELECT gibbonCourse.nameShort AS course, gibbonCourseClass.nameShort AS class, gibbonCourseClass.gibbonCourseClassID, gibbonCourseClass.gibbonCourseClassID, gibbonPlannerEntry.name AS lesson, gibbonPlannerEntryID, date, homework, homeworkSubmission FROM gibbonCourse JOIN gibbonCourseClass ON (gibbonCourse.gibbonCourseID=gibbonCourseClass.gibbonCourseID) JOIN gibbonCourseClassPerson ON (gibbonCourseClass.gibbonCourseClassID=gibbonCourseClassPerson.gibbonCourseClassID) JOIN gibbonPlannerEntry ON (gibbonCourseClass.gibbonCourseClassID=gibbonPlannerEntry.gibbonCourseClassID) WHERE (date<=:date1 AND date>=:date2) AND gibbonSchoolYearID=:gibbonSchoolYearID AND gibbonCourseClassPerson.gibbonPersonID=:gibbonPersonID AND role='Student' ORDER BY course, class, date DESC, timeStart";
-                        $resultSelect = $connection2->prepare($sqlSelect);
-                        $resultSelect->execute($dataSelect);
+                    $resultSelect = $container->get(CourseGateway::class)->selectLessonByStudent($session->get('gibbonSchoolYearID'), $values['gibbonPersonID']);
```

Comment:
This could be best in the PlannerEntryGateway.
```suggestion
                    $resultSelect = $container->get(CourseGateway::class)->selectPlannerLessonsByStudent($session->get('gibbonSchoolYearID'), $values['gibbonPersonID']);
```